nautobot/dcim/views.py

```diff
@@ -314,6 +327,147 @@ class LocationBulkDeleteView(generic.BulkDeleteView):
+
+        associated_object_id = obj.pk
+        associated_object_content_type = ContentType.objects.get_for_model(Location)
+        action = request.POST.get("action")
+        try:
+            with transaction.atomic():
+                if not has_perms(request.user, ["extras.add_contactassociation"]):
+                    raise PermissionDenied(
+                        "ObjectPermission extras.add_contactassociation is needed to perform this action"
+                    )
+                contact = None
+                team = None
+                if action == "create and assign new contact":
+                    if not has_perms(request.user, ["extras.add_contact"]):
+                        raise PermissionDenied("ObjectPermission extras.add_contact is needed to perform this action")
+                    contact = Contact(
+                        name=request.POST.get("name"),
+                        phone=request.POST.get("phone"),
+                        email=request.POST.get("email"),
+                    )
+                    contact.validated_save()
+                    # Trigger permission check
+                    Contact.objects.restrict(request.user, "view").get(pk=contact.pk)
+                elif action == "create and assign new team":
+                    if not has_perms(request.user, ["extras.add_team"]):
+                        raise PermissionDenied("ObjectPermission extras.add_team is needed to perform this action")
+                    team = Team(
+                        name=request.POST.get("name"),
+                        phone=request.POST.get("phone"),
+                        email=request.POST.get("email"),
+                    )
+                    team.validated_save()
+                    # Trigger permission check
+                    Team.objects.restrict(request.user, "view").get(pk=team.pk)
+                elif action == "use existing contact":
+                    contact = Contact.objects.restrict(request.user, "view").get(pk=request.POST.get("contact"))
+                elif action == "use existing team":
+                    team = Team.objects.restrict(request.user, "view").get(pk=request.POST.get("team"))
+                else:
+                    msg = f"Invalid action {action} passed from the form"
+                    logger.debug(msg)
+                    form.add_error(None, msg)
+                    return render(
+                        request,
+                        self.template_name,
+                        {
+                            "obj": obj,
+                            "obj_type": self.queryset.model._meta.verbose_name,
+                            "form": form,
+                            "return_url": self.get_return_url(request, obj),
+                            "editing": obj.present_in_database,
+                            **self.get_extra_context(request, obj),
+                        },
+                    )
```

Comment:
Maybe cleaner to handle this as another exception to raise and catch below so that we don't need the duplicate `render()` call here?